obj.js

```diff
@@ -425,59 +413,57 @@ export default class BaseProtocol {
     };
   }
 
-  async sendClassDescription(className, instance) {
-    if (this._localClasses.has(className)) {
-      return;
-    }
-    this._localClasses.add(className);
-
-    let proto;
-    if (instance) {
-      proto = Object.getPrototypeOf(instance);
-    }
-
-    let descr = {
-      className: className,
-      iterator: null,
-      functions: [],
-      getters: [],
-      properties: [],
-    };
+  getClassDescription(objDescrJSON) {
+    let classDescrJSON = [];
+    let instance = this.getLocalObject(objDescrJSON.idRemote);
+    while (true) {
+      let className = getClassName(instance);
```

Comment:
```suggestion
      if (!instance) {
        break;
      }
      let className = getClassName(instance);
```

I know that `getClassName()` defaults to "Object" when `instance` is null, but we shouldn't rely on the behavior of another function here in this case.

Can you see any other values of `.prototype` that we might encounter and that are not handled? To avoid infinitive loops, in error situations.

Reply:
It doesn't default to "Object" when `instance` is null, instead it does so when `proto` is null. In other words, we know that `proto` below is never null, and therefore neither is `instance`. I however would be happy to change line 419 to
```
    while (instance) { // should never fail due to class name check on previous iteration
```